### PR TITLE
Start supporting marking files

### PR DESCRIPTION
**Pull request by richfitz**
This is something that we will need for the orderly MVP - we want to mark files as immutable (this will affect things copied in as dependencies, resources, the script etc) and verify at the end of the run that nothing has changed them.  The proposed interface here will work like:

```r
  inputs <- c("orderly.yml", dat$script, dat$resources)
  outpack::outpack_packet_start(dst, name, id = id, root = root$outpack)
  outpack::outpack_packet_file_mark(inputs)
  outpack::outpack_packet_run(dat$script)
  status <- outpack::outpack_packet_file_list()
  # warn here about unknown files, then save the metadata and finish
```

Later on, we'll expand the mark function to support "ignored" so that we can use this to exclude files from the final packet. This is not implemented yet.

### CODE REVIEW

File: R/packet.R [hunk 1]
```diff
@@ -271,6 +274,59 @@ outpack_packet_add_custom <- function(application, data, schema = NULL) {
...
+##'
+##' @param files A character vector of relative paths
+##'
+##' @param status A status to mark the file with. Must be "immutable"
+##'   for now, later we will expand this.
+##'
+##' @return
+##' * `outpack_packet_file_mark` returns nothing
+##' * `outpack_packet_file_list` returns a [data.frame] with columns
+##'   `path` and `status` (`immutable`, `ignored`, `unknown`)
```
Comment (r-ash) [inline R/packet.R]: Status can't be `ignored` at moment?

File: tests/testthat/test-packet.R [hunk 2]
```diff
@@ -473,3 +473,102 @@ test_that("pre-prepared id can be used to start packet", {
...
+  root <- outpack_init(path, path_archive = "archive", use_file_store = TRUE)
+  on.exit(unlink(path, recursive = TRUE), add = TRUE)
+
+  path_src <- tempfile()
+  fs::dir_create(path_src)
+  on.exit(unlink(path_src, recursive = TRUE), add = TRUE)
+
+  path_src <- tempfile()
+  fs::dir_create(path_src)
+  on.exit(unlink(path_src, recursive = TRUE), add = TRUE)
```
Comment (r-ash) [inline tests/testthat/test-packet.R:491]: I see we are doing this in a few places now. Could add a helper function instead and use `withr::defer_parent` for the cleanup

**Review by r-ash — APPROVED** [review]
Looks great, couple of small comments but this looks good as is.

File: tests/testthat/test-packet.R (hunk 2 above)
Comment (richfitz) [reply tests/testthat/test-packet.R:491]: Good idea, logged this as https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-3298

File: R/packet.R (hunk 1 above)
Comment (richfitz) [reply R/packet.R]: good catch, thanks

**Review by hillalex — APPROVED** [review]

**Pull request merged**